text/2022-04-29-tls-support-for-httpchaos.md

```diff
@@ -0,0 +1,110 @@
+# TLS Support for HTTPChaos
+
+## Summary
+
+A solution to support TLS in HTTPChaos.
+
+## Motivation
+
+Currently, our HTTPChaos can only injects faults into unencrypted TCP connection.
+Once the clients and servers connect to others by TLS, we can do nothing in the middle.
+With the HTTPS protects more and more connections, we must figure out a solution to attack it.
+
+## Detailed Design
+
+### User-Provided Secrets
+
+The simplest solutions, mainly used in the case of server-side HTTPChaos, is providing private-key(the secret) by the user.
```

Comment:
> The simplest solutions, mainly used in the case of server-side HTTPChaos

I think there is no difference between server-side and client-side, we just hijack the TLS connection and do not even care anything about "trust things".

Reply:
I have the same opinion.

Reply:
Of course, there is no difference in implementation, but there should be differences in normal cases. User-provided secrets can resolve most use cases of server-side HTTPChaos.